projects/dotcms-field-elements/src/components/dot-key-value/dot-key-value.e2e.ts

```diff
@@ -1,168 +1,423 @@
+
+        it('should have filled required pristine', async () => {
+            element.setProperty('required', true);
+            element.setProperty('value', 'key|value,key2|value2');
+
+            await page.waitForChanges();
+            expect(element).toHaveClasses(dotTestUtil.class.filledRequiredPristine);
+        });
+
+        it('should have filled required touched when item is added', async () => {
+            element.setProperty('required', true);
+            const form = await getForm();
+            form.triggerEvent('add', {
+                detail: {
+                    key: 'some',
+                    value: 'test'
+                }
+            });
+            await page.waitForChanges();
+            expect(element).toHaveClasses(dotTestUtil.class.filledRequired);
+        });
+
+        it('should have filled required touched when one item is removed', async () => {
+            element.setProperty('value', 'key|value,llave|valor');
+            element.setProperty('required', true);
+            const list = await getList();
+            list.triggerEvent('delete', { detail: 0 });
+            await page.waitForChanges();
+            expect(element).toHaveClasses(dotTestUtil.class.filledRequired);
+        });
+    });
+
+    describe('@Props', () => {
+        describe('key-value-form attrs', () => {
+            it('should pass down valid props', async () => {
+                element.setAttribute('form-add-button-label', 'Button to the label');
+                element.setAttribute('form-key-placeholder', 'Key Placeholder');
+                element.setAttribute('form-value-placeholder', 'Value Placeholder');
+                element.setAttribute('form-key-label', 'Key Label');
+                element.setAttribute('form-value-label', 'Value Label');
+
                 await page.waitForChanges();
 
-                const tableRows = await page.findAll('table tr');
-                expect(tableRows.length).toBe(2);
-                expect(spyStatusChangeEvent).toHaveReceivedEventDetail({
-                    name: 'testName',
-                    status: {
-                        dotPristine: false,
-                        dotTouched: true,
-                        dotValid: true
-                    }
-                });
-                expect(spyValueChangeEvent).toHaveReceivedEventDetail({
-                    name: 'testName',
-                    fieldType: 'Key-Value',
-                    value: 'valueA|1,k|2'
-                });
+                const form = await getForm();
+                expect(form.getAttribute('add-button-label')).toBe('Button to the label');
+                expect(form.getAttribute('key-placeholder')).toBe('Key Placeholder');
+                expect(form.getAttribute('value-placeholder')).toBe('Value Placeholder');
+                expect(form.getAttribute('key-label')).toBe('Key Label');
+                expect(form.getAttribute('value-label')).toBe('Value Label');
```

Comment:
many expect in one it?

Reply:
Since I'm only passing down props I feels that this is a better approach. @alfredo-dotcms @hmoreras ?